### DISCUSSION TITLE
Refresh whitelist when detected change to whitelist file

### DISCUSSION THREAD

**Discussion by Player505** [itzg/docker-minecraft-server#2959] [Q&A]
### Enhancement Type

Improve an existing feature

### Describe the enhancement

Currently when adding players to the whitelist file provided to the `WHITELIST_FILE` environment variable the change isn't reflected till the container restarts which kinda sucks because to whitelist new players you have to basically kick any currently logged in players.

I propose using something akin to [incrond](https://www.cyberciti.biz/faq/linux-inotify-examples-to-replicate-directories/) to monitor changes in the whitelist file provided to the `WHITELIST_FILE` environment variable and when a change is detected running the relevant script based on the `EXISTING_WHITELIST_FILE` environment variable then sending an rcon command the server to reload the whitelist.

**Comment by itzg** [comment]
I prefer not to run additional processes inside of the container. The automation you described can be performed by a sidecar container via rcon.

**Comment by Player505** [comment]
That's understandable, I might see if I can make a sidecar container to do that

**Comment by itzg** [comment]
You might be able to draw some inspiration from the mc-backup container since it has similar cross container rcon interactions 

https://github.com/itzg/docker-mc-backup/blob/master/scripts/opt/backup-loop.sh

**Comment by twkonefal** [comment]
~~I'd like to jump in here and ask whether there are other solutions to authorization available that don't require discord or other external authentication services. Is it possible to disable the whitelist altogether and just use a db or other file outside of docker's compose configuration?~~

In fact, after some searching I saw there are plugins for velocity which allow for authorization management. I'm using [VelocityWhiteList](https://github.com/TISUnion/VelocityWhitelist/) and it's working out perfectly.
